src/connection/base.ts

```diff
@@ -224,6 +225,15 @@ export abstract class Connection {
     }
   }
 
+  private async throwErrorIfErrorBody(response: Response) {
+    // Hack, but looks like this is a limitation of the fetch API
+    // In order to read the body here and elesewhere, we need to clone the response
+    // since body can only be read once
+    const clonedResponse = response.clone();
+    const json = await clonedResponse.json();
```

Comment:
May be I do not understand Node.js enough but according to my understanding if not JSON response is accepted htere the lie `cloneResponse.json()` will throw parsing exception. In this case we will arrive to the catch block that will re-throw the same error that is not the "real" error but error of JSON parsing. Am I missing something?

Reply:
Fair point, added a try/catch for JSON parsing.